projects/Mallard/src/hooks/use-net-info.tsx

```diff
@@ -93,9 +93,16 @@ export class NetInfoStateContainer {
     }
 
     subscribe(fn: (state: NetInfoState) => void) {
-        this.subscribers.push(fn)
-        fn(this.state)
+        let shouldSubscribe = true
+        this.fetch()
+            .then(fn)
```

Comment:
We shouldn't call `fn` is `shouldSubscribe` has already switched to `false`, either.